src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,149 @@
+package duke;
+
+import duke.command.*;
+import duke.dukeexception.MissingDeadlineException;
+import duke.dukeexception.MissingTimingException;
+import duke.dukeexception.UnknownCommandException;
+import duke.task.Deadlines;
+import duke.task.Events;
+import duke.task.Task;
+import duke.task.ToDos;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import static java.lang.Boolean.parseBoolean;
+
+public class Parser {
+    private static DateTimeFormatter DATE_TIME_INPUT_FORMAT = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
+    private static DateTimeFormatter DATE_TIME_OUTPUT_FORMAT = DateTimeFormatter.ofPattern("MMM dd yyyy HHmm");
+
+    /**
+     * Method that takes in user input and parses through it to return an executable command.
+     *
+     * @param input String input that the user keys in.
+     * @return Command object that can be executed by duke.Duke.
+     * @throws UnknownCommandException When the input is invalid and is not executable.
+     */
+    public static Command parse(String input) throws UnknownCommandException {
+        try {
+            if (input.equals("bye")) {
+                return new ExitCommand();
+            }
+
+            if (input.equals("list")) {
+                return new ListCommand();
+            }
+            //logic to mark tasks with error handling
+            if (input.indexOf("mark") == 0) {
+                String substring = input.substring(5);
+                //Since first task is of index 0 in ArrayList
+                int taskIndex = Integer.parseInt(substring) - 1;
+                return new MarkCommand(taskIndex, true);
+            }
+            //logic to unmark tasks with error handling
+            if (input.indexOf("unmark") == 0) {
+                String substring = input.substring(7);
+                //Since first task is of index 0 in ArrayList
+                int taskIndex = Integer.parseInt(substring) - 1;
+                return new MarkCommand(taskIndex, false);
+            }
+            //logic to create duke.task.ToDos
+            if (input.indexOf("todo") == 0) {
+                String description = input.substring(5);
+                ToDos todo = new ToDos(description, false);
+                return new AddCommand(todo);
+            }
+            //logic to create duke.task.Deadlines
+            if (input.indexOf("deadline") == 0) {
+                //remove initial command
+                String sub = input.substring(9);
+                int timeIndex = sub.lastIndexOf("/by");
+                if (timeIndex == -1) {
+                    throw new MissingDeadlineException();
+                }
```

Comment:
Is there a reason why you use if else statements instead of switch statements? Maybe you can use enums for the different commands